protocol/xmrmaker/claim.go

```diff
@@ -234,63 +234,58 @@ func waitForClaimReceipt(
 	secret [32]byte,
 ) (*ethtypes.Receipt, error) {
 	const (
-		checkInterval = time.Second // time between transaction polls
-		maxWait       = time.Minute // max wait for the tx to be included in a block
-		maxNotFound   = 10          // max failures where the tx is not even found in the mempool
+		checkInterval = 15000 * time.Second // 1.5 seconds between poll attempts
+		maxPolls      = 10                  // We'll wait up to 15 seconds
 	)
 
-	start := time.Now()
-
-	var notFoundCount int
-	// wait for inclusion
-	for {
-		// sleep before the first check, b/c we want to give the tx some time to propagate
-		// into the node we're using
-		err := common.SleepWithContext(ctx, checkInterval)
-		if err != nil {
+	// The relayer can see the transaction as included in a block and send us
+	// the hash before our end sees it included in a block. The synchronization
+	// should happen significantly faster, but we allow a full 15 seconds before
+	// we give-up and decide that the relayer gave us a bad tx hash.
+	for i := 0; i < maxPolls; i++ {
+		receipt, err := ec.TransactionReceipt(ctx, txHash)
+		if err != nil && !errors.Is(err, ethereum.NotFound) {
 			return nil, err
 		}
-
-		_, isPending, err := ec.TransactionByHash(ctx, txHash)
+		// If err is still set, the error was ethereum.NotFound, which is returned
+		// by TransactionReceipt even if our endpoint sees the TX as pending.
 		if err != nil {
-			// allow up to 5 NotFound errors, in case there's some network problems
-			if errors.Is(err, ethereum.NotFound) && notFoundCount >= maxNotFound {
```

Comment:
I was wrong in my initial assessment. The check on what was formerly line 270 is correct, because it is `!isPending`. The actual problem was on this line, which should have been continuing the loop if `notFoundCount` was less than `maxNotFound`.